src/mono/browser/runtime/loader/icu.ts

```diff
@@ -46,6 +46,7 @@ export function getIcuResourceName (config: MonoConfig): string | null {
     if (config.resources?.icu && config.globalizationMode != GlobalizationMode.Invariant) {
         // TODO: when starting on sidecar, we should pass default culture from UI thread
         const culture = config.applicationCulture || (ENVIRONMENT_IS_WEB ? (globalThis.navigator && globalThis.navigator.languages && globalThis.navigator.languages[0]) : Intl.DateTimeFormat().resolvedOptions().locale);
+        config.applicationCulture = culture;
```

Comment:
nit: `config.applicationCulture ??= ENVIRONMENT_IS_WEB ? (globalThis.navigator && globalThis.navigator.languages && globalThis.navigator.languages[0]) : Intl.DateTimeFormat().resolvedOptions().locale`

Assuming that I understood this correctly and you are only setting this when the applicationCulture is not defined in the first place.